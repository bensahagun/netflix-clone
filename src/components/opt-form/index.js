import React from 'react';
import { Container, Input, Button, Break, Text } from './styles/opt-form';

export default function OptForm({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

OptForm.Input = function ({ ...restProps }) {
  return <Input {...restProps}></Input>;
};

OptForm.Button = function ({ children, ...restProps }) {
  return (
    <Button {...restProps}>
      {children} <img src='/images/icons/chevron-right.png' alt='Try now' />
    </Button>
  );
};

OptForm.Text = function ({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

OptForm.Break = function ({ ...restProps }) {
  return <Break {...restProps} />;
};
