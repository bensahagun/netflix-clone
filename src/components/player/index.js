import React, { useState, useContext, createContext } from 'react';
import ReactDOM from 'react-dom';
import { Container, Button, FeatureButton, Overlay, Inner, Close } from './styles/player';

export const PlayerContext = createContext();

export default function Player({ children, ...restProps }) {
  const [showPlayer, setShowPlayer] = useState(false);

  return (
    <PlayerContext.Provider value={{ showPlayer, setShowPlayer }}>
      <Container {...restProps}>{children}</Container>
    </PlayerContext.Provider>
  );
}

Player.Video = function PlayerVideo({ src, ...restProps }) {
  const { showPlayer, setShowPlayer } = useContext(PlayerContext);

  return showPlayer
    ? ReactDOM.createPortal(
        <Overlay onClick={() => setShowPlayer(false)} {...restProps}>
          <Inner>
            <video id='netflix-player' controls autoPlay='autoplay'>
              <source src={src} type='video/mp4' />
            </video>
          </Inner>
          <Close />
        </Overlay>,
        document.body
      )
    : null;
};

Player.Button = function PlayerButton({ ...restProps }) {
  // eslint-disable-next-line no-unused-vars
  const { showPlayer, setShowPlayer } = useContext(PlayerContext);

  return (
    <Button {...restProps} onClick={() => setShowPlayer((showPlayer) => !showPlayer)}>
      Play
    </Button>
  );
};

Player.FeatureButton = function PlayerFeatureButton({ ...restProps }) {
  // eslint-disable-next-line no-unused-vars
  const { showPlayer, setShowPlayer } = useContext(PlayerContext);

  return (
    <FeatureButton {...restProps} onClick={() => setShowPlayer((showPlayer) => !showPlayer)}>
      Play Now
    </FeatureButton>
  );
};
