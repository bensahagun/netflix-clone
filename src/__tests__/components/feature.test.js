import React from 'react';
import { render } from '@testing-library/react';
import { Feature } from '../../components';

describe('<Feature/>', () => {
  it('render the <Feature/> with populated data', () => {
    const { container, getByText } = render(
      <Feature>
        <Feature.Title>Unlimited films, TV programmes and more.</Feature.Title>
        <Feature.SubTitle>Watch Anywhere. Cancel at any time.</Feature.SubTitle>
      </Feature>
    );

    expect(getByText('Unlimited films, TV programmes and more.')).toBeTruthy();
    expect(getByText('Watch Anywhere. Cancel at any time.')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('render the <Feature/> with just the title', () => {
    const { container, queryByText } = render(
      <Feature>
        <Feature.Title>Unlimited films, TV programmes and more.</Feature.Title>
      </Feature>
    );

    expect(queryByText('Unlimited films, TV programmes and more.')).toBeTruthy();
    expect(queryByText('Watch Anywhere. Cancel at any time.')).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('render the <Feature/> with just the subtitle', () => {
    const { container, queryByText } = render(
      <Feature>
        <Feature.SubTitle>Watch Anywhere. Cancel at any time.</Feature.SubTitle>
      </Feature>
    );

    expect(queryByText('Unlimited films, TV programmes and more.')).toBeFalsy();
    expect(queryByText('Watch Anywhere. Cancel at any time.')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
