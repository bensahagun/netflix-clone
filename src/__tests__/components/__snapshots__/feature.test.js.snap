// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Feature/> render the <Feature/> with just the subtitle 1`] = `
<div
  class="feature__Container-sc-6swrp6-0 lfRLWn"
>
  <h2
    class="feature__SubTitle-sc-6swrp6-2 bUoOlM"
  >
    Watch Anywhere. Cancel at any time.
  </h2>
</div>
`;

exports[`<Feature/> render the <Feature/> with just the title 1`] = `
<div
  class="feature__Container-sc-6swrp6-0 lfRLWn"
>
  <h1
    class="feature__Title-sc-6swrp6-1 hhTbUE"
  >
    Unlimited films, TV programmes and more.
  </h1>
</div>
`;

exports[`<Feature/> render the <Feature/> with populated data 1`] = `
<div
  class="feature__Container-sc-6swrp6-0 lfRLWn"
>
  <h1
    class="feature__Title-sc-6swrp6-1 hhTbUE"
  >
    Unlimited films, TV programmes and more.
  </h1>
  <h2
    class="feature__SubTitle-sc-6swrp6-2 bUoOlM"
  >
    Watch Anywhere. Cancel at any time.
  </h2>
</div>
`;
